cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -261,6 +261,20 @@ if (NOT CMAKE_FIND_ROOT_PATH_MODE_PACKAGE)
   set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
 endif()
 
+set(_em_pkgconfig_libdir "${EMSCRIPTEN_SYSROOT}/local/lib/pkgconfig" "${EMSCRIPTEN_SYSROOT}/lib/pkgconfig")
+if("${CMAKE_VERSION}" VERSION_LESS "3.20")
+  file(TO_NATIVE_PATH "${_em_pkgconfig_libdir}" _em_pkgconfig_libdir)
+  if(CMAKE_HOST_UNIX)
+    string(REPLACE ";" ":" _em_pkgconfig_libdir "${_em_pkgconfig_libdir}")
+    string(REPLACE "\\ " " " _em_pkgconfig_libdir "${_em_pkgconfig_libdir}")
+  endif()
+else()
+  cmake_path(CONVERT "${_em_pkgconfig_libdir}" TO_NATIVE_PATH_LIST _em_pkgconfig_libdir)
+endif()
+set(ENV{PKG_CONFIG_LIBDIR} "${_em_pkgconfig_libdir}")
+set(ENV{PKG_CONFIG_PATH} "$ENV{EM_PKG_CONFIG_PATH}") # Stops caller's PKG_CONFIG_PATH from propagating.
```

Comment:
I think we should forget about  EM_PKG_CONFIG_PATH.  I'm not sure why use it ever had.

If somebody explicitly sets PKG_CONFIG_PATH when building an emscripten project I don't see why should ignore it.  That seems like a deliberate act.

Reply:
This behaviour was based on [tools.building.get_building_env()](https://github.com/emscripten-core/emscripten/blob/main/tools/building.py#L153).

I agree that it seems a bit out of place, but that would introduce inconsistency in how caller's variables are used across tools. If it's acceptable for you then I'm happy to do it.

Or should I update `get_building_env`? For example: create a backwards-compatible behaviour that prioritises `EM_PKG_CONFIG_PATH` but also allows for `PKG_CONFIG_PATH`, and then implement it consistently in`get_building_env` and here in toolchain.

Reply:
I think you should ignore `EM_PKG_CONFIG_PATH` in this patch and we should try to remove it from `get_building_env` as separate change.

Reply:
Alright, removed the line in question.